src/main/java/org/opensearch/neuralsearch/search/query/HybridCollectorManager.java

```diff
@@ -211,6 +306,61 @@ private TotalHits getTotalHits(
         return new TotalHits(maxTotalHits, relation);
     }
 
+    private TopDocs getNewTopFieldDocs(final TotalHits totalHits, final List<TopFieldDocs> topFieldDocs, final SortField sortFields[]) {
```

Comment:
nit: I feel like we need a new abstraction for getting these top docs, we can do it later in separate PR